src/elliptic/curves/curve_ristretto.rs

```diff
@@ -360,6 +348,5 @@ impl Zeroize for RistrettoPoint {
     fn zeroize(&mut self) {
         unsafe { ptr::write_volatile(&mut self.ge, PK::default()) };
         atomic::fence(atomic::Ordering::SeqCst);
-        atomic::compiler_fence(atomic::Ordering::SeqCst);
```

Comment:
Afaik compiler_fence doesn't emit any machine code, it's just a hint for a compiler. Is it unnecessary anyway?

Reply:
I looked up what zeroize crate does ([here](https://docs.rs/zeroize/1.4.1/src/zeroize/lib.rs.html#241-249), [here](https://docs.rs/zeroize/1.4.1/src/zeroize/lib.rs.html#511-513)) — it do `atomic::compiler_fence`, and skips `atomic::fence`. But I'm not sure what's the difference

Reply:
Argh, I meant to leave the `compiler_fence` and remove the `fence`, the CPU fence is very extreme and doesn't really related to this (there's no threads here because we have `&mut self`)

Reply:
(And as you showed that's also how Zeroize does it)